src/main/kotlin/no/nav/tilleggsstonader/sak/vedtak/barnetilsyn/beregning/TilsynBeregningUtils.kt

```diff
@@ -28,11 +28,11 @@ object TilsynBeregningUtil {
      * listOf(StønadsperiodeDto(fom = 01.01.24, tom=10.01.24), StønadsperiodeDto(fom = 20.01.24, tom=31.01.24)) deles opp i 2 innenfor samme måned:
      * jan -> listOf(StønadsperiodeDto(fom = 01.01.24, tom = 10.01.24), StønadsperiodeDto(fom = 20.01.24, tom = 31.01.24))
      */
-    fun List<StønadsperiodeBeregningsgrunnlag>.tilÅrMåned(): Map<YearMonth, List<StønadsperiodeBeregningsgrunnlag>> =
+    fun <P> List<P>.tilÅrMåned(): Map<YearMonth, List<P>> where P : Periode<LocalDate>, P : KopierPeriode<P> =
         this
             .flatMap { stønadsperiode ->
                 stønadsperiode.splitPerMåned { måned, periode ->
-                    periode.copy(
+                    periode.medPeriode(
                         fom = maxOf(periode.fom, måned.atDay(1)),
                         tom = minOf(periode.tom, måned.atEndOfMonth()),
                     )
```

Comment:
Er det noe spesiell grunn til å gjøre den om til generisk? 
Ettersom denne er unik for tilsyn barn, så kan det være tilstrekkelig å la den se ut som tidligere, men bruke `VedtaksperiodeBeregningsgrunnlag` i stedet for `StønadsperiodeBeregningsgrunnlag` og ikke gjøre den om til generisk